test/specs/orgCreationAndAuthentication.e2e.ts

```diff
@@ -0,0 +1,267 @@
+import {
+  EnvironmentSettings
+} from '../environmentSettings';
+import * as utilities from '../utilities';
+
+describe('Org Creation and Authentication', async () => {
+  const tempProjectName = 'TempProject-OrgCreationAndAuth';
+  let tempFolderPath: string;
+  let projectFolderPath: string;
+  let prompt: QuickOpenBox | InputBox;
+  let scratchOrgAliasName: string;
+
+  step('Set up the testing environment', async () => {
+    tempFolderPath = getTempFolderPath();
+    projectFolderPath = path.join(tempFolderPath, tempProjectName);
+
+    // Remove the project folder, just in case there are stale files there.
+    if (fs.existsSync(projectFolderPath)) {
+      utilities.removeFolder(projectFolderPath);
+      await utilities.pause(1);
+    }
+
+    // Now create the temp folder.  It should exists but create the folder if it is missing.
+    if (!fs.existsSync(tempFolderPath)) {
+      await utilities.createFolder(tempFolderPath);
+      await utilities.pause(1);
+    }
+  });
+
+  step('Run SFDX: Create Project', async () => {
+    const workbench = await browser.getWorkbench();
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Create Project', 5);
+    // Selecting "SFDX: Create Project" causes the extension to be loaded, and this takes a while.
+
+    // Select the "Standard" project type.
+    let quickPicks = await prompt.getQuickPicks();
+    expect(quickPicks).not.toBeUndefined();
+    expect(quickPicks.length).toBeGreaterThanOrEqual(1);
+    expect(quickPicks.length).toEqual(3);
+    expect(await quickPicks[0].getLabel()).toEqual('Standard');
+    expect(await quickPicks[1].getLabel()).toEqual('Empty');
+    expect(await quickPicks[2].getLabel()).toEqual('Analytics');
+    await prompt.selectQuickPick('Standard');
+    await utilities.pause(1);
+
+    // Enter "TempProject-OrgCreationAndAuth" for the project name.
+    await prompt.setText(tempProjectName);
+    await utilities.pause(1);
+
+    // Press Enter/Return.
+    await prompt.confirm();
+
+    // Set the location of the project.
+    const input = await prompt.input$;
+    await input.setValue(tempFolderPath);
+    await utilities.pause(1);
+
+    // Click the OK button.
+    await utilities.clickFilePathOkButton();
+
+    // Verify the project was created and was loaded.
+    const sidebar = workbench.getSideBar();
+    const content = sidebar.getContent();
+    const treeViewSection = await content.getSection(tempProjectName.toUpperCase());
+    expect(treeViewSection).not.toEqual(undefined);
+
+    const forceAppTreeItem = await treeViewSection.findItem('force-app') as DefaultTreeItem;
+    expect(forceAppTreeItem).not.toEqual(undefined);
+
+    await forceAppTreeItem.expand();
+
+    // Yep, we need to wait a long time here.
+    await utilities.pause(10);
+  });
+
+  step('Run SFDX: Authorize a Dev Hub', async () => {
+    // This is essentially the "SFDX: Authorize a Dev Hub" command, but using the CLI and an auth file instead of the UI.
```

Comment:
The choice to use the CLI here is that we can't use the VS Code command due to needing that auth file, right?

Reply:
Yes, exactly